libraries/botbuilder-core/tests/telemetryMiddleware.test.js

```diff
@@ -3,97 +3,164 @@
 
 const assert = require('assert');
 const { TestAdapter, TelemetryLoggerMiddleware, ActivityTypes } = require('../');
-const { TeamsChannelData } = require('../../botframework-schema/');
 
-describe(`TelemetryMiddleware`, function () {
+class OverrideReceiveLogger extends TelemetryLoggerMiddleware {
+    async onReceiveActivity(activity) {
+        this.telemetryClient.trackEvent({
+            name: TelemetryLoggerMiddleware.botMsgReceiveEvent,
+            properties: {
+                'foo': 'bar',
+                'ImportantProperty': 'ImportantValue'
+            }
+        });
+        this.telemetryClient.trackEvent({
+            name: 'MyReceive',
+            properties: await this.fillReceiveEventProperties(activity, { 'conversationName': 'OVERRIDE' })
+        });
+    }
+}
+
+class OverrideSendLogger extends TelemetryLoggerMiddleware {
+    async onSendActivity(activity) {
+        this.telemetryClient.trackEvent({
+            name: TelemetryLoggerMiddleware.botMsgSendEvent,
+            properties: {
+                'foo': 'bar',
+                'ImportantProperty': 'ImportantValue'
+            }
+        });
+        this.telemetryClient.trackEvent({
+            name: 'MySend',
+            properties: await this.fillSendEventProperties(activity)
+        });
+    }
+}
+
+class OverrideUpdateDeleteLogger extends TelemetryLoggerMiddleware {
+    async onUpdateActivity() {
+        this.telemetryClient.trackEvent({
+            name: TelemetryLoggerMiddleware.botMsgUpdateEvent,
+            properties: {
+                'foo': 'bar',
+                'ImportantProperty': 'ImportantValue'
+            }
+        });
+    }
+    async onDeleteActivity() {
+        this.telemetryClient.trackEvent({
+            name: TelemetryLoggerMiddleware.botMsgDeleteEvent,
+            properties: {
+                'foo': 'bar',
+                'ImportantProperty': 'ImportantValue'
+            }
+        });
+    }
+}
+
+function createReply(activity, text, locale = null) {
+    return {
+        type: ActivityTypes.Message,
+        from: { id: activity.recipient.id, name: activity.recipient.name },
+        recipient: { id: activity.from.id, name: activity.from.name },
+        replyToId: activity.id,
+        serviceUrl: activity.serviceUrl,
+        channelId: activity.channelId,
+        conversation: { isGroup: activity.conversation.isGroup, id: activity.conversation.id, name: activity.conversation.name },
+        text: text || '',
+        locale: locale || activity.locale
+    };
+}
+
+describe(`TelemetryMiddleware`, function() {
     this.timeout(5000);
-    it(`telemetry should log send and receive activities`, function (done) {
+    it(`telemetry should log send and receive activities`, function(done) {
         var callCount = 0;
         var telemetryClient = {
             trackEvent: (telemetry) => {
-                assert(telemetry, 'telemetry is null');
```

Comment:
These changes look big, but they're just wrapping all the assertions in a `try/catch` so that:

```js
try {
  // assertions
  done();
} catch(err) {
  done(err); // this is the only way I could get the test to fail
}
```